internal/daemon/plugin/vendorplugin.go

```diff
@@ -132,21 +133,37 @@ func WithVsp(template_vars VspTemplateVars) func(*GrpcPlugin) {
 	}
 }
 
-func (gp *GrpcPlugin) deployVsp() {
+func (gp *GrpcPlugin) deployVsp() error {
 	vspImage := gp.vsp.VendorSpecificPluginImage
 
-	if vspImage != "" {
-		gp.log.Info("Deploying VSP", "vspImage", vspImage, "command", gp.vsp.Command, "args", gp.vsp.Args)
-		err := render.ApplyAllFromBinData(gp.log, "vsp-ds", gp.vsp.ToMap(), binData, gp.k8sClient, nil, scheme.Scheme)
-		if err != nil {
-			gp.log.Error(err, "Failed to start vendor plugin container", "vspImage", gp.vsp.VendorSpecificPluginImage)
-		}
-	} else {
+	// It is not mandatory that a vsp image is provided. If not, we can assume this will be handled by the user and still return a GrpcClient
+	if vspImage == "" {
 		gp.log.Info("WARNING: VSP Image not set, skipping vendor plugin container startup")
+		return nil
+	}
+
+	// Retrieve the Dpu Operator Config which owns the Dpu Daemonset so we can ensure the vsp shares the same owner reference.
+	namespace := os.Getenv("NAMESPACE")
+	dpuOperatorConfigName := os.Getenv("DPU_OPERATOR_CONFIG_NAME")
```

Comment:
Can we get rid of this?

Reply:
Now that I know we can assume the config name, yes. I suppose we could assume the namespace as well, but typically this is something I see provided in the downward API, so I figured it would be safest to consume it.

Reply:
I think we might need to create a struct called DownwardApi and add functions to fetch anything that's coming from the downward api. Why? Because code that calls os.Getenv is untestable. It's tedious to refactor code that is sprinkled with os.Getenv, and we should aim to avoid writing any code that ties ourselves to the os.Getenv. 

That being said, I think we don't need the downward api here at all, because we can just assume the hard coded name.